### PR TITLE
updated to newer Checkbox

### PR DESCRIPTION
**Pull request by akp111**
## Explanation

As per the comments on this PR https://github.com/MetaMask/metamask-extension/pull/20198, addressing the issue https://github.com/MetaMask/metamask-extension/issues/20164, I have broken down the changes into smaller PR. This one addresses replacing the checkbox in `ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js`

* Part: https://github.com/MetaMask/metamask-extension/issues/20163
### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/092034b4-4fc7-4bfe-88ac-3f760c9ccad7

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/98530b82-db26-4143-aa0a-21a9cdda356b

image
image

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Thanks for your contribution @akp111, could you please add some before and after screencasts of the UI to show no visual regressions?

**Comment by akp111** [comment]
Hey @georgewrmarshall , I have added that. Also I have raised other PRs to replace the checkbox

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.stories.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      <AdvancedGasFeePopoverContextProvider>
+        {story()}
+      </AdvancedGasFeePopoverContextProvider>
+    </GasFeeContextProvider>
+  </Provider>],
+};
+
+export const DefaultStory = () => <AdvancedGasFeeDefaults />;
+
+DefaultStory.storyName = 'Default';
```
Comment (georgewrmarshall) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.stories.js:47]: Screenshot 2023-08-01 at 2 17 44 PM

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution @akp111. I push some small updates and a story for `AdvancedGasFeeDefaults` so we can easily update it in future and have updated the PR description with screen casts of before/after

**Review by georgewrmarshall — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]

**Pull request merged**